controllers/storagecluster/reconcile.go

```diff
@@ -448,9 +448,11 @@ func (r *StorageClusterReconciler) reconcilePhases(
 			statusutil.SetErrorCondition(&instance.Status.Conditions, reason, message)
 
 			if instance.Status.Phase != statusutil.PhaseOnboarding {
-				// if the error was due to skipped storage classes
+				// if the error was due to skipped storage classes or
+				// due to StorageProfile list retrieval failure
 				// instead of error phase we will set it to progressing
-				if strings.Contains(returnErr.Error(), storageClassSkippedError) {
+				if strings.Contains(returnErr.Error(), storageClassSkippedError) ||
```

Comment:
- IMHO, mixing errors and strings isn't a good practice
- Not to refactor existing code, can we at least not follow this in this context and compare errors with `errors.Is`